### PR TITLE
Performance Improvements

### PR DESCRIPTION
**Pull request by armiol**
This PR improves the performance of the `core-java` modules basing on the speed of `DeliveryTest` integration tests.

Highlights:

- Some of the most called but redundant `vBuild()` calls switched to `build()`.
- The most often validated messages are made `FieldAwareMessage`s and their `readValue(..)` was implemented to avoid reflective access during the validation.
- Guava's `Cache` was used to reduce the CPU and memory footprint for the commonly asked/created data (e.g. `BoundedContextName` for a raw `Class`, which was an expensive lookup).

### CODE REVIEW

File: core/src/main/java/io/spine/core/CommandIdMixin.java [hunk 1]
```diff
@@ -20,16 +20,26 @@
...
+interface CommandIdMixin extends SignalId, FieldAwareMessage, CommandIdOrBuilder {
 
     @Override
     default String value() {
         return getUuid();
     }
+
+    @Override
+    default Object readValue(Descriptors.FieldDescriptor field) {
+        if(0 == field.getIndex()) {
```
Comment (dmdashenkov) [inline core/src/main/java/io/spine/core/CommandIdMixin.java]: Please apply autoformatting.

File: core/src/main/java/io/spine/core/CommandMixin.java [hunk 2]
```diff
@@ -88,4 +91,20 @@ default boolean isScheduled() {
         }
         return false;
     }
+
+    @Override
+    default Object readValue(Descriptors.FieldDescriptor field) {
+        switch (field.getIndex()) {
+            case 0:
```
Comment (dmdashenkov) [inline core/src/main/java/io/spine/core/CommandMixin.java]: Can we use named constants for field numbers instead?

File: core/src/main/java/io/spine/core/EventIdMixin.java [hunk 3]
```diff
@@ -20,16 +20,26 @@
...
+interface EventIdMixin extends SignalId, FieldAwareMessage, EventIdOrBuilder {
 
     @Override
     default String value() {
         return getValue();
     }
+
+    @Override
+    default Object readValue(Descriptors.FieldDescriptor field) {
+        if(0 == field.getIndex()) {
```
Comment (dmdashenkov) [inline core/src/main/java/io/spine/core/EventIdMixin.java]: Please apply autofiormatting.

File: server/src/main/java/io/spine/server/aggregate/AggregateEndpoint.java [hunk 4]
```diff
@@ -177,7 +180,7 @@ private DispatchOutcome applyProducedEvents(A aggregate, DispatchOutcome command
             event.getContextBuilder()
                  .setVersion(signalVersion);
         }
-        return correctedCommandOutcome.vBuild();
+        return correctedCommandOutcome.build();
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/aggregate/AggregateEndpoint.java]: Please use `buildPartial` here and in similar cases. We should explicitly state the absence of validation.

File: server/src/main/java/io/spine/server/dispatch/ProducedCommandsMixin.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import io.spine.annotation.GeneratedMixin;
+import io.spine.validate.FieldAwareMessage;
+
+/**
+ * A mixin interface for the {@link ProducedCommands} message type.
+ */
+@GeneratedMixin
+public interface ProducedCommandsMixin extends ProducedCommandsOrBuilder, FieldAwareMessage {
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/dispatch/ProducedCommandsMixin.java]: Please make generated mixins package-private when possible.

File: server/src/main/proto/spine/server/dispatch/dispatching.proto [hunk 6]
```diff
@@ -133,6 +139,7 @@ message Interruption {
 
 // Ignored signal handling.
 message Ignore {
+    option (is).java_type = "IgnoreMixin";
```
Comment (dmdashenkov) [inline server/src/main/proto/spine/server/dispatch/dispatching.proto]: Let's consider making this a convention and removing the need to add the option if the message type implements an interface named in the same way as the type itself but with the `Mixin` suffix.

If I remember correctly, we already pass the whole classpath to our `protoc` plugin. We might as well check if there is such an interface.

Although, this is, most likely, not a v1.0 feature.

Comment (dmdashenkov) [reply server/src/main/proto/spine/server/dispatch/dispatching.proto]: Also, generating messages to implement `FieldAwareMessage` is fairly easy at this point.

File: server/src/main/java/io/spine/server/aggregate/AggregateEndpoint.java (hunk 4 above)
Comment (armiol) [reply server/src/main/java/io/spine/server/aggregate/AggregateEndpoint.java]: I view this differently. Let's discuss in person.

Comment (alexander-yevsyukov) [reply server/src/main/java/io/spine/server/aggregate/AggregateEndpoint.java]: The absence of `v` prefix does this.

File: core/src/main/java/io/spine/core/CommandMixin.java (hunk 2 above)
Comment (armiol) [reply core/src/main/java/io/spine/core/CommandMixin.java]: We would have to depend on the descendants (i.e. on the classes, which are marked by this mixin interface), as the `...OrBuilder` has no such notion. So, unfortunately, no.

File: server/src/main/java/io/spine/server/dispatch/ProducedCommandsMixin.java (hunk 5 above)
Comment (armiol) [reply server/src/main/java/io/spine/server/dispatch/ProducedCommandsMixin.java]: Done.

File: server/src/main/proto/spine/server/dispatch/dispatching.proto (hunk 6 above)
Comment (armiol) [reply server/src/main/proto/spine/server/dispatch/dispatching.proto]: As discussed in person, we would rather invest in a generation of a fully-blown in-place validation.

**Pull request merged**